modules/rational-modeline.el

```diff
@@ -0,0 +1,158 @@
+;;;; rational-modeline.el --- rational module for setting up the modeline  -*- lexical-binding: t; -*-
+
+;; Copyright (C) 2022
+;; SPDX-License-Identifier: MIT
+
+;; Authors: System Crafters Community, David "System Crafters" Wilson, Erik Lundstedt
+
+;; Commentary
+
+;;
+
+;;; Code:
+
+(defcustom test 'nil
+  "something here"
+  :group 'rational
+  :set (lambda (sym val)
+         (set-default sym val)
+         (when (equal test t)
+           (counsel-find-file)
+           )
+         )
+  )
+
+
+;;(when (string-equal rational-modeline-type "GNU")
+;;TODO make it look as good as possible without external packages
+;;  )
+
+
+
+
+(defun doom-modeline-type ()
+  (straight-use-package 'doom-modeline)
```

Comment:
None of these functions have either the `rational-` prefix, nor a docstring. Please add some documentation.